components/Dialog.jsx

```diff
@@ -61,3 +61,19 @@ Dialog.Footer.propTypes = {
   children: PropTypes.node,
   className: PropTypes.string
 }
+
+Dialog.Overlay = ({ children, className, show, ...props }) => (
+  <div
+    className={classNames('cui__dialog__overlay', { 'is-visible': show }, className)}>
+    <div
+      className={classNames('cui__dialog__wrapper')}>
+      {children}
+    </div>
+  </div>
+)
+
+Dialog.Overlay.propTypes = {
+  children: PropTypes.node,
```

Comment:
Nope. It can work without the overlay. Some applications live inside an iframe that is used in a dialog, so they can't have the full chrome of the overlay inside.